monai/data/dataset.py

```diff
@@ -848,10 +848,14 @@ def _compute_cache(indices=None):
 
     def disable_share_memory_cache(self):
         """
-        If the cache content is multiprocessing share memory list, convert it to a regular ptython list.
-        Because multiprocessing ProxyList is not supported for the GPU caching, may need to explicitly diasble it.
+        If the cache content is multiprocessing share memory list, convert it to a regular python list.
+        Because multiprocessing ProxyList is not supported for the GPU caching, may need to explicitly disable it.
 
         """
+        if self._is_dist:
+            torch.distributed.barrier()
+            if torch.distributed.get_rank() == 0:
+                time.sleep(0.5)  # rank 0 should exit after all the other ranks, as the cache was broadcast from rank 0
```

Comment:
I don't think so, if it works then the following should work as well:
```py
def disable_share_memory_cache(self):
    self._cache = list(self._cache)
    if self._is_dist:
        dist.barrier()
```
this doesn't because the rank 0 list manager is already closed before the barrier...